src/Generator/AttributeGenerator.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Laminas\Code\Generator;
+
+use Laminas\Code\Generator\AttributeGenerator\AttributeAssembler;
+use Laminas\Code\Generator\AttributeGenerator\AttributePrototype;
+use Laminas\Code\Generator\AttributeGenerator\AttributeWithArgumentsAssembler;
+use Laminas\Code\Generator\AttributeGenerator\SimpleAttributeAssembler;
+use ReflectionAttribute;
+use ReflectionClass;
+
+final class AttributeGenerator implements GeneratorInterface
+{
+    private array $assemblers;
+
+    private function __construct(AttributeAssembler ...$assembler)
+    {
+        $this->assemblers = $assembler;
+    }
+
+    public function generate(): string
+    {
+        $generatedAttributes = array_map(fn(AttributeAssembler $attributeAssembler) => $attributeAssembler->assemble(),
+            $this->assemblers,
+        );
+
+        return implode(AbstractGenerator::LINE_FEED, $generatedAttributes);
+    }
+
+    public static function fromPrototype(AttributePrototype ...$attributePrototype): self
+    {
+        $assemblers = [];
+        
+        foreach ($attributePrototype as $prototype) {
+            $assemblers[] = self::negotiateAssembler($prototype);
+        }
+
+        return new self(...$assemblers);
+    }
+
+    public static function fromReflection(ReflectionClass $reflectionClass): self
+    {
+        $attributes = $reflectionClass->getAttributes();
+        $assemblers = [];
+
+        foreach ($attributes as $attribute) {
+            $assembler = self::negotiateAssembler($attribute);
+
+            $assemblers[] = $assembler;
+        }
+
+        return new self(...$assemblers);
+    }
+
+    public static function fromArray(array $definitions): self
```

Comment:
I decided to drop all `::fromArray()` methods as per #153

This API will be dropped once I have the green light on #153